pkg/allowedpaths/cmd.go

```diff
@@ -0,0 +1,85 @@
+package allowedpaths
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// CommandWithLookup should be used when the full path to the command is not known and it is
+// likely to be found in PATH.
+func CommandWithLookup(name string, arg ...string) (*exec.Cmd, error) {
+	fullPathToCommand, err := exec.LookPath(name)
+	if err != nil {
+		return nil, fmt.Errorf("looking up path to %s: %w", name, err)
+	}
+
+	return CommandWithPath(fullPathToCommand, arg...)
+}
+
+// CommandWithPath should be used when the full path to the command is known, or unlikely to be
+// found in PATH.
+func CommandWithPath(fullPathToCommand string, arg ...string) (*exec.Cmd, error) {
+	fullPathToCommand = filepath.Clean(fullPathToCommand)
+
+	if err := pathIsAllowed(fullPathToCommand); err != nil {
+		return nil, fmt.Errorf("path is not allowed: %w", err)
+	}
+
+	return exec.Command(fullPathToCommand, arg...), nil //nolint:forbidigo
+}
+
+// CommandContextWithLookup should be used when the full path to the command is not known and it is
+// likely to be found in PATH.
+func CommandContextWithLookup(ctx context.Context, name string, arg ...string) (*exec.Cmd, error) {
+	fullPathToCommand, err := exec.LookPath(name)
+	if err != nil {
+		return nil, fmt.Errorf("looking up path to %s: %w", name, err)
+	}
+
+	return CommandContextWithPath(ctx, fullPathToCommand, arg...)
+}
+
+// CommandContextWithPath should be used when the full path to the command is known, or unlikely to be
+// found in PATH.
+func CommandContextWithPath(ctx context.Context, fullPathToCommand string, arg ...string) (*exec.Cmd, error) {
+	fullPathToCommand = filepath.Clean(fullPathToCommand)
+
+	if err := pathIsAllowed(fullPathToCommand); err != nil {
+		return nil, fmt.Errorf("path is not allowed: %w", err)
+	}
+
+	return exec.CommandContext(ctx, fullPathToCommand, arg...), nil //nolint:forbidigo
+}
+
+// pathIsAllowed validates the path to the command against our allowlist.
+func pathIsAllowed(fullPathToCommand string) error {
+	cmdName := strings.ToLower(filepath.Base(fullPathToCommand))
```

Comment:
I can't decide if this is clever, or too clever. And maybe I want to understand the callers.

Reply:
(I moved this conversation to slack)